src/PowerShellEditorServices.Host/CodeLens/PesterCodeLensProvider.cs

```diff
@@ -15,71 +15,90 @@ namespace Microsoft.PowerShell.EditorServices.CodeLenses
+        /// <param name="scriptFile">The script file the Pester symbol comes from.</param>
+        /// <returns>All CodeLenses for the given Pester symbol.</returns>
+        private CodeLens[] GetPesterLens(
             PesterSymbolReference pesterSymbol,
             ScriptFile scriptFile)
         {
-            var clientCommands = new ClientCommand[]
+            var codeLensResults = new CodeLens[]
             {
-                new ClientCommand(
-                    "PowerShell.RunPesterTests",
-                    "Run tests",
-                    new object[]
-                    {
-                        scriptFile.ClientFilePath,
-                        false, // Don't debug
-                        pesterSymbol.TestName,
-                    }),
+                new CodeLens(
+                    this,
+                    scriptFile,
+                    pesterSymbol.ScriptRegion,
+                    new ClientCommand(
+                        "PowerShell.RunPesterTests",
+                        "Run tests",
+                        new object[] { scriptFile.ClientFilePath, false /* No debug */, pesterSymbol.TestName })),
 
-                new ClientCommand(
-                    "PowerShell.RunPesterTests",
-                    "Debug tests",
-                    new object[]
-                    {
-                        scriptFile.ClientFilePath,
-                        true, // Run in debugger
-                        pesterSymbol.TestName,
-                    }),
+                new CodeLens(
+                    this,
+                    scriptFile,
+                    pesterSymbol.ScriptRegion,
+                    new ClientCommand(
+                        "PowerShell.RunPesterTests",
+                        "Debug tests",
+                        new object[] { scriptFile.ClientFilePath, true /* Run in debugger */, pesterSymbol.TestName })),
             };
 
-            return
-                clientCommands.Select(
-                    command =>
-                        new CodeLens(
-                            this,
-                            scriptFile,
-                            pesterSymbol.ScriptRegion,
-                            command));
+            return codeLensResults;
         }
 
+        /// <summary>
+        /// Get all Pester CodeLenses for a given script file.
+        /// </summary>
+        /// <param name="scriptFile">The script file to get Pester CodeLenses for.</param>
+        /// <returns>All Pester CodeLenses for the given script file.</returns>
         public CodeLens[] ProvideCodeLenses(ScriptFile scriptFile)
         {
-            var symbols =
-                this.symbolProvider
-                    .ProvideDocumentSymbols(scriptFile);
+            var lenses = new List<CodeLens>();
+            foreach (SymbolReference symbol in _symbolProvider.ProvideDocumentSymbols(scriptFile))
```

Comment:
This only calls `ProvideDocumentSymbols` once and iterates through the result, right?

Reply:
I think so, yes. Why do you ask?

Reply:
spoke to @SeeminglyScience about this. We're good